launch_ros/launch_ros/utilities/evaluate_parameters.py

```diff
@@ -54,7 +54,23 @@ def evaluate_parameter_dict(
             if isinstance(value[0], Substitution):
                 # Value is a list of substitutions, so perform them to make a string
                 evaluated_value = perform_substitutions(context, list(value))
-                evaluated_value = yaml.safe_load(evaluated_value)
+                yaml_evaluated_value = yaml.safe_load(evaluated_value)
+                evaluated_type = type(yaml_evaluated_value)
+                if evaluated_type in (bool, int, float, str):
+                    evaluated_value = yaml_evaluated_value
+                # TODO(ivanpauno): Maybe, add better checking of other sequence types.
+                if isinstance(yaml_evaluated_value, list):
+                    # If it is a list with dissimilar types, let the string without converting.
+                    last_subtype = None
+                    dissimilar_types = False
+                    for subvalue in yaml_evaluated_value:
+                        subtype = type(subvalue)
+                        if last_subtype is not None and last_subtype != subtype:
+                            dissimilar_types = True
+                            break
+                        last_subtype = subtype
+                    if not dissimilar_types and subtype in (bool, int, float, str):
+                        evaluated_value = tuple(yaml_evaluated_value)
```

Comment:
You mentioned that you were implicitly making this a tuple as it was in other places, but does anyone know if this is the right thing to do? I guess it might, but why can you not just do `evaluated_value = yaml_evaluated_value` or `evaluated_value = copy.copy(yaml_evaluated_value)`?

Reply:
I guess you're copying the code from the `elif isinstance(value[0], Sequence):` block.

Reply:
I think that doing `copy.copy` or converting it to a `tuple` is not needed.

Reply:
When I wrote this I made it return tuples because I wanted the output of `normalize_parameters` to be an immutable thing on the composable node description. However, that didn't work out because they include dictionaries, and there isn't a built in immutable version of that. I never undid evaluating sequences as `tuple`. It probably could be anything as long as the output type is consistent.

Reply:
Any is fine by me, I just wanted to bring it up for discussion.